torch/package/exporter.py

```diff
@@ -198,6 +228,7 @@ def _get_source_of_module(self, module: types.ModuleType) -> str:
         return ''.join(result)
 
     def require_module_if_not_provided(self, module_name: str, dependencies=True):
+        module_name = self._demangle(module_name)
```

Comment:
I feel like this demangling is coming too late. The only mangled properties are the `__name__` fields of modules. I feel like the only safe place to put a demangle call is directly in the same function that looks up `__name__` from the module. Otherwise, you get this big separation between when a mangled name is extract and when it is used. The _only_ place we should want mangled names is in the `__name__` field so that Python infra doesn't get confused about module lookup.